awx/ui_next/src/screens/Template/TemplateList/TemplateList.jsx

```diff
@@ -215,28 +226,32 @@ class TemplatesList extends Component {
                     itemName={i18n._(t`Template`)}
                   />,
                   canAdd && (
-                    <Dropdown
+                    <div
+                      ref={node => {
+                        this.node = node;
+                      }}
```

Comment:
I like that you're using ref to hang onto the DOM node.  We need to take a step back and determine how we plan to keep track of this dropdown node and the [empty list state dropdown](https://github.com/ansible/awx/blob/devel/awx/ui_next/src/screens/Template/TemplateList/TemplateList.jsx#L257). Since the empty list dropdown isn't attached to a ref, the toggle functionality is broken when there are no items in the list. 

My one other concern, `handleAddToggle` will fire on every page click. In the future, we should refactor this so that the event listeners are only added while the dropdown is open.